Update for dashboard

c-o-l-o-r opened the pull request:
@villanuevawill

villanuevawill commented on bounties_api/std_bounties/models.py:
```diff
@@ -225,6 +225,7 @@ class Fulfillment(models.Model):
     fulfiller_email = models.CharField(max_length=128, blank=True)
     fulfiller_githubUsername = models.CharField(max_length=128, blank=True)
     fulfiller_address = models.CharField(max_length=128, blank=True)
+    fulfiller_profile_image = models.CharField(max_length=256, blank=True)
```
@c-o-l-o-r  why are you adding fulfiller_profile_image to the model? This isn't the right way to go about doing this. Lets talk :)

villanuevawill replied:
There's a user relation... :) Just include it in the serializer.

c-o-l-o-r commented:
@villanuevawill @codeluggage this PR should be ready, please take look at it to make sure everything is in order! Here is what I did:

* fixed incorrect function signature (no `from_user`) for profile updates in the notification client
* added isDirty flag to determine users who need to have their profile image updated
* updated skills endpoint to serve RankedSkills rather than just skills
* added a count for total fulfillments posted to a user's bounty as a user stat

codeluggage commented on bounties_api/user/views.py:
```diff
@@ -117,6 +123,16 @@ def get(self, request, public_address):
+        ).get('fulfillments_count__sum', None)
+
+        if total_fulfillments_on_bounties:
+            total_fulfillments_on_bounties = total_fulfillments_on_bounties
```
I am not quite following this if check - does something special happen when assigning `total_fulfillments_on_bounties` to `total_fulfillments_on_bounties` here?

c-o-l-o-r replied:
This is good feedback, I need to remove that. Thanks @codeluggage 😄

villanuevawill replied:
Also confused here :)

c-o-l-o-r commented on bounties_api/std_bounties/client_helpers.py:
```diff
@@ -172,17 +173,29 @@ def map_user_data(data_hash, public_address):
     return prune(user)
 
 
+def isProfileImageDirty(ipfs_hash, public_address):
```
It is definitely weirdly worded. I'm open to better suggestions! I was trying to use similar terminology used in caching system, but it isn't widely used in our codebase

codeluggage replied:
Yeah, `dirty` has more understood connotations. To get the same meaning using other words, it would need to both be `modified` but also have the concept of being modified elsewhere. `isProfileImageModifiedUpstream` or something, which is seriously verbose. Going with `dirty` is probably best.

c-o-l-o-r commented on bounties_api/std_bounties/filters.py:
```diff
@@ -28,6 +28,7 @@ class Meta:
         fields = {
             'fulfiller': ['exact'],
             'bounty': ['exact'],
+            'bounty__issuer_address': ['exact'],
```
I think the double underscore is django convention. I wasn't sure how to search on the `issuer_address` any other ways, but please let me know if there is a better way.

codeluggage replied:
No that definitely sounds right. Learning a little more python and django every day 😀

c-o-l-o-r commented on bounties_api/user/migrations/0011_user_isprofileimagedirty.py:
```diff
@@ -0,0 +1,20 @@
+    operations = [
+        migrations.AddField(
+            model_name='user',
+            name='isProfileImageDirty',
```
not that I cant see, and it definitely drives me up the wall haha

codeluggage replied:
😭

villanuevawill replied:
haha - this is the most annoying thing to me... @codeluggage @c-o-l-o-r we should stick to camel case 100% in python. The only reason snake case exists in this codebase was originally to support the integrity of the field names on the bounty model. I now regret this decision as it caused significant inconsistencies in the readability of the codebase. PLEASE, PLEASE, please, all future elements make snake case.

villanuevawill replied:
oh my goodnes... I meant snake case UGH!!! hahahaah

c-o-l-o-r replied:
so snake case for variables AND funcitons in python?

villanuevawill commented on bounties_api/user/views.py:
```diff
@@ -117,6 +117,16 @@ def get(self, request, public_address):
         fulfiller_fulfillment_acceptance = None if not user_fulfillments.count() else (
             user_fulfillments.filter(accepted=True).count() / user_fulfillments.count())
 
+        total_fulfillments_on_bounties = user_bounties.annotate(
```
@c-o-l-o-r @codeluggage  pretty sure there's a WAY easier way to do this:
`user_bounties.fulfillments.count()`
Does that not work?

c-o-l-o-r replied:
I tried this a few other things, but was unable to get it to work. I feel like there should be something easier, but I could not figure it out 😢.

villanuevawill commented on bounties_api/std_bounties/client_helpers.py:
```diff
@@ -172,17 +173,29 @@ def map_user_data(data_hash, public_address):
     plucked_data = pluck(data, user_data_keys)
     plucked_social = pluck(data.get('social', {}), user_social_keys)
 
+    profileDirectoryHash = data.get('profilePhoto', {}).get('fileDirectoryHash', None)
```
@c-o-l-o-r PLEASE, please, please snake case :)

c-o-l-o-r replied:
you got it captain

villanuevawill commented:
@c-o-l-o-r LGTM!! Merge it in :)
